### PR TITLE
buffs embed pulling with hemostats, allows wirecutters to pull embeds too

### PR DESCRIPTION
**Pull request by Hatterhat**
## About The Pull Request
- Wirecutters or tools with wirecutter behaviors are now valid for plucking embeds.
- Pluck speed no longer **starts** at 2.5 seconds, which is a pretty dang long time, especially if you have bad embed RNG. I'll do the math and run some more tests in the morning.
- Wirecutters have a speed malus in regards to plucking embeds. I should probably make it worse to account for, like, jaws of life or something.
- Plucking embeds with wirecutters now hurts! It hurts way less than ripping it out with your hands, but it still hurts!

For comparison's sake, bare-handed throwing star removal compared to possible tools.
image

## Why It's Good For The Game
Embeds kinda suck to deal with. This is intentional - I get that.

However, hemostat pulling is kind of... kind of bad. Awful, really. 2.5 seconds is a lot of time. I know it's not supposed to be the best option, but if you've got a tool, I'd at least like to think it'd be slightly less bad than shoving your fingers into your wound?

## Changelog

:cl:
balance: Pulling embedded items e.g. shrapnel with hemostats is now a lot faster, and scales appropriately with toolspeed.
balance: You can now pull embedded items with wirecutters, at a speed penalty.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
Making pull out slow is dumb, no matter how ghetto. At a certain level there's just no point to wasting your time with it when you can use those seconds to actually seek medical attention. It should have drawbacks and downsides that are not related to a low effort 10 second delay that makes it irrelevant.

**Comment by norsvenska** [comment]
why is it faster to remove embedded objects with alien wirecutters compared to alien hemostat? shouldn't it be the other way around?

File: code/datums/components/embedded.dm [hunk 1]
```diff
@@ -286,8 +293,11 @@
 			to_chat(victim, span_danger("[user] fails to pluck [weapon] from your [limb.plaintext_zone]."))
 		return
 
-	to_chat(user, span_notice("You successfully pluck [weapon] from [victim]'s [limb.plaintext_zone]."))
-	to_chat(victim, span_notice("[user] plucks [weapon] from your [limb.plaintext_zone]."))
+	to_chat(user, span_notice("You [tweezer_safe ? "safely" : "unsafely, but"] successfully pluck [weapon] from [victim]'s [limb.plaintext_zone]."))
```
Comment (Ghommie) [inline code/datums/components/embedded.dm:296]:
```suggestion
	to_chat(user, span_notice("You [tweezer_safe ? "" : "unsafely, but"] successfully pluck [weapon] from [victim]'s [limb.plaintext_zone]."))
```
I don't think the `safely` adverb is needed, plus you've forgot to add a conjuction between it and `successfully` anyway.

File: code/datums/components/embedded.dm [hunk 2]
```diff
@@ -286,8 +293,11 @@
...
+	to_chat(victim, span_notice("[user] [tweezer_safe ? "safely" : span_warning("painfully")] plucks [weapon] from your [limb.plaintext_zone]."))
```
Comment (Ghommie) [inline code/datums/components/embedded.dm:297]:
```suggestion
	to_chat(victim, span_notice("[user] [tweezer_safe ? "" : span_warning("painfully")] plucks [weapon] from your [limb.plaintext_zone]."))
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by Hatterhat** [comment]
> why is it faster to remove embedded objects with alien wirecutters compared to alien hemostat? shouldn't it be the other way around?

funny thing is this only happens because wirecutters have toolspeed 0.1 while the hemostats have toolspeed 0.25, so the math is just always in the wirecutters' favor

**Pull request merged**